src/stackedtime/sim_nr.jl

```diff
@@ -187,33 +206,37 @@ function damping_br81(; delta::Real=0.1, rateK::Real=10.0, lambda_min::Real=1e-5
         ::Union{Nothing,Factorization,AbstractMatrix{Float64}}=nothing,
         ::Union{Nothing,AbstractVector{Float64}}=nothing
     )
-        # @printf "    it=%d, λ=%g, nF=%g\n" it λ nF
-        it < 1 && (bigK = 0.0; return true, 1.0)
+        # Initialization step
+        it < 1 && (bigK = 0.0; return true, λ_max)
+        
         if nF2_it != it
-            # first time we're called this iteration 
+            # First time we're called in this iteration
             nF2 = norm(F, 2)  # store the residual 
             nF2_it = it
             return false, calc_λ()
         end
+        
         if (1 - δ * λ) * nF2 < norm(F, 2)
-            # test failed => reject and try smaller λ
+            # If test failed, decrease step size
             if bigK == 0.0
                 bigK = 1.0
             else
-                bigK = rateK * bigK
+                bigK *= rateK  # Increase `bigK` slower to prevent excessive reductions in λ
             end
             λ = calc_λ()
             if λ > λ_min
                 return false, λ
             else
-                # λ too small
                 verbose && @warn "Linesearch failed."
                 return true, λ_min
             end
         else
-            # lower bigK for next iteration ...
-            bigK = bigK / rateK
-            # ... and accept given λ 
+            # Lower `bigK` more aggressively when convergence is happening
+            bigK /= sqrt(rateK)
+
+            # If λ is near the lower bound for many steps, slowly increase it
+            λ = min(λ * λ_growth, λ_max)
```

Comment:
It's not quite clear to me why we need to grow lambda here? The br algorithm already increases lambda when it lowers bigK.